packages/tests/tests/wait_for_remote_peer.node.spec.ts

```diff
@@ -39,7 +39,7 @@ describe("Wait for remote peer", function () {
     await waku1.dial(multiAddrWithId);
     await delay(1000);
     await waitForRemotePeer(waku1, [Protocols.Relay]);
-    const peers = waku1.relay.getMeshPeers();
+    const peers = Array.from(waku1.relay.getMeshPeers().values()).flat();
```

Comment:
```suggestion
    const peers = waku1.relay.getMeshPeers().get(DefaultPubSubTopic);
```

The test is too forgiving IMO.

Also, I wonder if it would make sense to have `getMeshPeer` take a `PubSubTopic` as argument so only peers for the shard we are interested in are returned.

Reply:
I mean that it does not take the pubsub topic in account. We could be connecting to another peer but for the wrong pubsub topic and the test would still pass.
The test can be more _strict_ here easily.